app/src/main/java/org/oppia/android/app/viewmodel/BUILD.bazel

```diff
@@ -32,8 +32,8 @@ kt_android_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
-        artifact("javax.inject:javax.inject:1"),
         artifact("androidx.appcompat:appcompat"),
         artifact("androidx.lifecycle:lifecycle-extensions:2.2.0"),
+        artifact("javax.inject:javax.inject:1"),
```

Comment:
Is this correct? The version (1) seems wrong, inject isn't listed in the WORKSPACE, and I actually thought this came through the JSR annotations dependency, instead.

Reply:
@BenHenning I've tried using `javax.annotation:javax.annotation-api:jar` but the build fails. 
Actually if found `javax.inject:javax.inject:1` from the External dependencies.
![image](https://user-images.githubusercontent.com/54740946/107481362-0c4b4480-6ba4-11eb-8bfc-fdc571a32462.png)

Reply:
@FareesHussain can you try this one: https://github.com/oppia/oppia-android/blob/e7cc00098de00dd45f11ab6975b0f399a4c13393/app/BUILD.bazel#L513.

Reply:
@BenHenning Tried this but it doesn't work
```
ERROR: /home/farees/dev/oppia/oppia-android/app/src/main/java/org/oppia/android/app/viewmodel/BUILD.bazel:27:19: KotlinCompile //app/src/main/java/org/oppia/android/app/viewmodel:view_model_provider_kt { kt: 2, java: 0, srcjars: 0 } for k8 failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/io_bazel_rules_kotlin/src/main/kotlin/builder ... (remaining 1 argument(s) skipped)
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:5:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:6:14: error: unresolved reference: inject
import javax.inject.Provider
             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:13:46: error: unresolved reference: Inject
class ViewModelBridgeFactory<V : ViewModel> @Inject constructor(
                                             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:14:34: error: unresolved reference: Provider
  private val viewModelProvider: Provider<V>
                                 ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:7:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:13:41: error: unresolved reference: Inject
class ViewModelProvider<V : ViewModel> @Inject constructor(
                                        ^
Compilation failure: compile phase failed:
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:5:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:6:14: error: unresolved reference: inject
import javax.inject.Provider
             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:13:46: error: unresolved reference: Inject
class ViewModelBridgeFactory<V : ViewModel> @Inject constructor(
                                             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:14:34: error: unresolved reference: Provider
  private val viewModelProvider: Provider<V>
                                 ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:7:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:13:41: error: unresolved reference: Inject
class ViewModelProvider<V : ViewModel> @Inject constructor(
                                        ^
INFO: Elapsed time: 31.847s, Critical Path: 16.28s
INFO: 211 processes: 14 internal, 188 linux-sandbox, 9 worker.
FAILED: Build did NOT complete successfully

```

Reply:
@FareesHussain can you try syncing to head & using ``//third_party:javax_annotation_javax_annotation-api_jar``? I'm still trying to figure out where the dependency is coming from. :)

Reply:
Getting this error now

```

ERROR: /home/farees/dev/oppia/oppia-android/app/src/main/java/org/oppia/android/app/viewmodel/BUILD.bazel:27:19: no such package 'third_party': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
 - /home/farees/dev/oppia/oppia-android/third_party and referenced by '//app/src/main/java/org/oppia/android/app/viewmodel:view_model_provider_base'
ERROR: Analysis of target '//app/src/main/java/org/oppia/android/app/viewmodel:view_model_provider_base' failed; build aborted: Analysis failed
INFO: Elapsed time: 22.577s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (16 packages loaded, 26 targets configured)
    Fetching @kt_java_stub_template; fetching
    Fetching @local_config_cc; Restarting.
    Fetching @maven; Restarting.
```

Reply:
@fsharpasharp Can you have a look at this.

Reply:
@FareesHussain you will have to replace all artifact dependencies with the corresponding third_party wrapper.

Reply:
@fsharpasharp Can you be a bit more specific. What changes do I have to make here?

Reply:
For example,

`artifact("androidx.lifecycle:lifecycle-extensions:2.2.0"),`

becomes

`"//third_party:androidx_lifecycle_lifecycle-extensions",`

Reply:
@fsharpasharp sorry for the delay. 
I did try this. But this doesn't seem to be working.
I get this error when I run the following command
`bazel build :all`
```
INFO: Analyzed 9 targets (1 packages loaded, 13 targets configured).
INFO: Found 9 targets...
ERROR: /home/farees/dev/oppia/oppia-android/app/src/main/java/org/oppia/android/app/viewmodel/BUILD.bazel:27:19: KotlinCompile //app/src/main/java/org/oppia/android/app/viewmodel:view_model_provider_kt { kt: 2, java: 0, srcjars: 0 } for k8 failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/io_bazel_rules_kotlin/src/main/kotlin/builder ... (remaining 1 argument(s) skipped)
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:5:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:6:14: error: unresolved reference: inject
import javax.inject.Provider
             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:13:46: error: unresolved reference: Inject
class ViewModelBridgeFactory<V : ViewModel> @Inject constructor(
                                             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:14:34: error: unresolved reference: Provider
  private val viewModelProvider: Provider<V>
                                 ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:7:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:13:41: error: unresolved reference: Inject
class ViewModelProvider<V : ViewModel> @Inject constructor(
                                        ^
Compilation failure: compile phase failed:
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:5:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:6:14: error: unresolved reference: inject
import javax.inject.Provider
             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:13:46: error: unresolved reference: Inject
class ViewModelBridgeFactory<V : ViewModel> @Inject constructor(
                                             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelBridgeFactory.kt:14:34: error: unresolved reference: Provider
  private val viewModelProvider: Provider<V>
                                 ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:7:14: error: unresolved reference: inject
import javax.inject.Inject
             ^
/home/farees/.cache/bazel/_bazel_farees/7de6a1daffd6876688d829bf5721b2d3/execroot/__main__/app/src/main/java/org/oppia/android/app/viewmodel/ViewModelProvider.kt:13:41: error: unresolved reference: Inject
class ViewModelProvider<V : ViewModel> @Inject constructor(
                                        ^
INFO: Elapsed time: 0.531s, Critical Path: 0.40s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully
```